rollup/sync_service.go

```diff
@@ -293,147 +293,155 @@ func (s *SyncService) Stop() error {
-				continue
-			}
-			log.Debug("Applying transaction", "index", i)
-			err = s.maybeApplyTransaction(tx)
-			if err != nil {
-				log.Error("Cannot apply transaction", "msg", err)
-			}
-			s.SetLatestIndex(&i)
+		log.Debug("Applying transaction", "index", i)
+		err = s.maybeApplyTransaction(tx)
+		if err != nil {
+			log.Error("could not apply transaction", "error", err)
 		}
-		time.Sleep(s.pollInterval)
+		s.SetLatestIndex(&i)
 	}
+
+	return nil
 }
 
 func (s *SyncService) SequencerLoop() {
 	log.Info("Starting Sequencer Loop", "poll-interval", s.pollInterval, "timestamp-refresh-threshold", s.timestampRefreshThreshold)
 	for {
-		// Only the sequencer needs to poll for enqueue transactions
-		// and then can choose when to apply them. We choose to apply
-		// transactions such that it makes for efficient batch submitting.
-		// Place as many L1ToL2 transactions in the same context as possible
-		// by executing them one after another.
-		// TODO: break this routine out into a function so that lock
-		// management is more simple. For now, be sure to unlock before
-		// each outer continue
 		s.txLock.Lock()
-		latest, err := s.client.GetLatestEnqueue()
+		err := s.sequence()
 		if err != nil {
-			log.Error("Cannot get latest enqueue")
-			s.txLock.Unlock()
-			time.Sleep(s.pollInterval)
-			continue
+			log.Error("Could not sequence", "error", err)
 		}
-		// This should never happen unless the backend is empty
-		if latest == nil {
-			log.Debug("No enqueue transactions found")
-			s.txLock.Unlock()
-			time.Sleep(s.pollInterval)
-			continue
-		}
-		// Compare the remote latest queue index to the local latest
-		// queue index. If the remote latest queue index is greater
-		// than the local latest queue index, be sure to ingest more
-		// enqueued transactions
-		var start uint64
-		if s.GetLatestEnqueueIndex() == nil {
-			start = 0
-		} else {
-			start = *s.GetLatestEnqueueIndex() + 1
-		}
-		end := *latest.GetMeta().QueueIndex
+		s.txLock.Unlock()
+		time.Sleep(s.pollInterval)
+	}
+}
 
-		log.Info("Polling enqueued transactions", "start", start, "end", end)
-		for i := start; i <= end; i++ {
-			enqueue, err := s.client.GetEnqueue(i)
-			if err != nil {
-				log.Error("Cannot get enqueue in loop", "index", i, "message", err)
-				continue
-			}
+func (s *SyncService) sequence() error {
+	// Only the sequencer needs to poll for enqueue transactions
+	// and then can choose when to apply them. We choose to apply
+	// transactions such that it makes for efficient batch submitting.
+	// Place as many L1ToL2 transactions in the same context as possible
+	// by executing them one after another.
+	latest, err := s.client.GetLatestEnqueue()
+	if err != nil {
+		return err
+	}
 
-			if enqueue == nil {
-				log.Debug("No enqueue transaction found")
-				break
-			}
+	// This should never happen unless the backend is empty
+	if latest == nil {
+		return errors.New("No enqueue transactions found")
```

Comment:
We don't want to return an error here because it will cause errors to be logged when starting up a fresh node with no L1 to L2 transactions sent yet. I think logging on `Debug` level and returning `nil` makes sense

Reply:
https://github.com/ethereum-optimism/go-ethereum/pull/289/commits/7484140dcbbd5683ebf96fc6f91f6eda83e302a4